client/src/components/InterviewHeader.js

```diff
@@ -1,31 +1,62 @@
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import { useStyles } from '../themes/theme';
 import Grid from '@material-ui/core/Grid';
 import LanguageSelectMenu from './LanguageSelectMenu';
+
+import { useHistory, useParams } from 'react-router-dom';
+import { store } from '../context/store';
+import SocketContext from '../context/socket';
 import { CustomButton } from '../components/Buttons';
-import { useHistory } from 'react-router-dom';
 
-const InterviewHeader = ({ language, setLanguage }) => {
+const InterviewHeader = ({ language, setLanguage, userData }) => {
   const classes = useStyles();
   const history = useHistory();
+  const {
+    state: {
+      user: { firstName, lastName },
+    },
+  } = useContext(store);
+
+  const socket = useContext(SocketContext);
 
   const handleLanguageChange = (event) => {
     setLanguage(event.target.value);
-  }
+  };
+  const { id: roomId } = useParams();
+
+  const [first, setFirst] = useState('');
+
+  useEffect(() => {
+    socket.emit('get_partner_name', { firstName, lastName, roomId });
+    return () => {};
+  }, [firstName, lastName, socket, roomId]);
+  useEffect(() => {
+    socket.on('set_partner_name', ({ firstName, lastName }) => {
+      setFirst(firstName + ' ' + lastName);
+    });
+    return () => {};
+  }, [socket]);
```

Comment:
We definitely want this data coming from the get interviews API call and to be passed as props to the header